src/server/scripts/Spells/spell_rogue.cpp

```diff
@@ -612,23 +618,46 @@ class spell_rog_stealth : public SpellScriptLoader
         {
             PrepareAuraScript(spell_rog_stealth_AuraScript);
 
-            bool Validate(SpellInfo const* /*spellInfo*/) override
-            {
-                return ValidateSpellInfo(
-                {
-                    SPELL_ROGUE_MASTER_OF_SUBTLETY_PASSIVE,
-                    SPELL_ROGUE_MASTER_OF_SUBTLETY_DAMAGE_PERCENT,
-                    SPELL_ROGUE_MASTER_OF_SUBTLETY_PERIODIC,
-                    SPELL_ROGUE_SANCTUARY,
-                    SPELL_ROGUE_STEALTH_STEALTH_AURA,
-                    SPELL_ROGUE_STEALTH_SHAPESHIFT_AURA
-                });
-            }
+          //bool Validate(SpellInfo const* /*spellInfo*/) override
+          //{
+          //    return ValidateSpellInfo(
+          //    {
+          //        SPELL_ROGUE_MASTER_OF_SUBTLETY_PASSIVE,
+          //        SPELL_ROGUE_MASTER_OF_SUBTLETY_DAMAGE_PERCENT,
+          //        SPELL_ROGUE_MASTER_OF_SUBTLETY_PERIODIC,
+          //        SPELL_ROGUE_SANCTUARY,
+          //        SPELL_ROGUE_STEALTH_STEALTH_AURA,
+          //        SPELL_ROGUE_STEALTH_SHAPESHIFT_AURA,
+          //        SPELL_ROGUE_NIGHTSTALKER_AURA,
+          //        SPELL_ROGUE_NIGHTSTALKER_DAMAGE_DONE,
+          //        SPELL_ROGUE_SHADOW_FOCUS_AURA,
+          //        SPELL_ROGUE_SHADOW_FOCUS_COST_PCT,
+          //        SPELL_ROGUE_STEALTH_SUBTERFUGE,
+          //        SPELL_ROGUE_STEALTH_SUBTERFUGE_EFFECT
+          //     });
+          //}
 
             void HandleEffectApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
             {
                 Unit* target = GetTarget();
+                if (Unit* target = GetTarget())
+                {
+                    if (GetSpellInfo()->Id != SPELL_ROGUE_STEALTH_SUBTERFUGE_EFFECT)
```

Comment:
this will never work, because the spell is not bound to this script

Reply:
Sorry I dont know to bound it.